src/threading/MsgThread.cc

```diff
@@ -227,6 +225,83 @@ MsgThread::MsgThread() : BasicThread(), queue_in(this, nullptr), queue_out(nullp
 
     io_source = new detail::IOSource(this);
 
+    cnt_sent_in_metric = telemetry_mgr->CounterInstance("zeek", "msg_thread_msgs_sent_in", {{"thread_name", Name()}},
+                                                        "Number of messages sent into thread");
+    cnt_sent_out_metric = telemetry_mgr->CounterInstance("zeek", "msg_thread_msgs_sent_out", {{"thread_name", Name()}},
+                                                         "Number of messages sent from thread");
+    pending_in_metric = telemetry_mgr->GaugeInstance("zeek", "msg_thread_msgs_pending_in", {{"thread_name", Name()}},
+                                                     "Number of pending messages sent into thread", "",
+                                                     [this]() -> prometheus::ClientMetric {
```

Comment:
The captured `this` in these is even harder to follow than previous instances since we don't have a clear global we could reference. If e.g., each `MsgThread` had a unique ID we could check whether the instance is still valid by capturing the ID and then going via the global `thread_mgr` (checked), but we don't have one. Do you have any other ideas for making this less fragile?

Reply:
And it blows up, as shown above. Think it would be better to collect metrics in the manager and return for alive threads, only.